### PR TITLE
Refactor graph pkg

### PR DESCRIPTION
**Pull request by timwangmusic**
## Description
Remove graph package

## Solution
* Move min priority queue code to planner package where it is used
* Upgrade Go version to 1.18

## Testing
- [x] Integration testing on Heroku staging
- [x] Added new unit tests

## Checks
- [x] Have you removed commented code?
- [x] Have you used gofmt to format your code?

### CODE REVIEW

File: planner/min_priority_queue.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+func (pq MinPriorityQueue) Less(i, j int) bool {
+	return pq[i].Key <= pq[j].Key
+}
+
+func (pq MinPriorityQueue) Swap(i, j int) {
+	pq[i], pq[j] = pq[j], pq[i]
+}
+
+func (pq *MinPriorityQueue) Push(item interface{}) {
```
Comment (rwangsc18) [inline planner/min_priority_queue.go:17]: Does the new data structure `MinPriorityQueue` keep the min-heap data structure? for example, the first element is always the minimum?

Comment (timwangmusic) [reply planner/min_priority_queue.go:17]: It's not a new data structure. Just renamed it.

Comment (rwangsc18) [reply planner/min_priority_queue.go:17]: I see, thanks.

**Review by rwangsc18 — APPROVED** [review]
LGTM

**Pull request merged**